test/common/upstream/health_checker_impl_test.cc

```diff
@@ -2018,7 +2158,6 @@ TEST_F(TcpHealthCheckerImplTest, Timeout) {
   add_uint8(response, 1);
   read_filter_->onData(response, false);
 
-  EXPECT_CALL(*connection_, close(_));
```

Comment:
why is this gone?

Reply:
By default reuse_connection is true, the timeout event won't close the connection. I don't understand how it was working before this change.

Reply:
I've added this back